qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp

```diff
@@ -412,7 +412,9 @@ std::size_t IndirectFittingModel::numberOfWorkspaces() const {
 }
 
 std::size_t IndirectFittingModel::getNumberOfSpectra(std::size_t index) const {
-  return m_fittingData[index]->numberOfSpectra();
+  if (index < m_fittingData.size())
+    return m_fittingData[index]->numberOfSpectra();
+  return 0;
```

Comment:
If one gives this function an invalid index, should the user be aware that this happened? Perhaps one should log a warning and/or throw an exception?